sdk/common/src/test/java/io/opentelemetry/sdk/internal/ComponentRegistryTest.java

```diff
@@ -8,54 +8,70 @@
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
+import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.sdk.common.InstrumentationScopeInfo;
 import org.junit.jupiter.api.Test;
 
 class ComponentRegistryTest {
 
-  private static final String INSTRUMENTATION_NAME = "test_name";
-  private static final String INSTRUMENTATION_VERSION = "version";
+  private static final String NAME = "test_name";
+  private static final String VERSION = "version";
+  private static final String SCHEMA_URL = "http://schema.com";
+  private static final Attributes ATTRIBUTES = Attributes.builder().put("k1", "v1").build();
   private final ComponentRegistry<TestComponent> registry =
       new ComponentRegistry<>(TestComponent::new);
 
   @Test
-  void scopeName_MustNotBeNull() {
-    assertThatThrownBy(() -> registry.get(null, "version"))
+  void get_NameAndAttributesMustNotByNull() {
```

Comment:
```suggestion
  void get_NameAndAttributesMustNotBeNull() {
```

Reply:
Why have these tests that do double duty? I'd say let's split these up into separate small tests.

Reply:
The double duty is really just the set of invalid argument test cases. There's a fair amount of precedence for grouping invalid args tests together in this repo (for [example](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractGrpcTelemetryExporterTest.java#L601-L632)).

I think I'll keep the tests together but rename the test to be `get_InvalidArgs()`.

Reply:
Yeah ok, that helps, and I understand the precedence. I don't think it's great practice tho, I'd love us to not perpetuate it. 🤷🏻‍♂️

Reply:
> I don't think it's great practice tho

I actually prefer grouping similar tests into a single test function. It can make it really easy to succinctly express the intended behavior the code. In certain cases, splitting each scenario into a separate adds keystrokes and makes the code harder to understand. Illegal arguments is one good example, and here's another [example](https://github.com/open-telemetry/opentelemetry-java/pull/4700/files#diff-301b1eb1bf3254c1c1ecaca01c8377184accbed10cc6030fa44541bf27848099R45-R76).

Reply:
Yup, like all things -- pros and cons. You forgot that separate methods runs slower too. 😁 
When something fails in a test that's testing several things, I think it becomes much harder to find out which condition is at fault. Furthermore, the first failure might hide the fact that several conditions are broken.